Automattic/Gravatar-SDK-Android
# Add Dokka Versioning and Keep generated dokka history

Closes #133 

### Description

I did a few things in this PR:
- Updated build files to use dokka-versioning plugin that adds a dropdown menu in the dokka menu to browse version history.
- Generated 0.3.0 docs with the versioning plugin and added them to the git repository
- Added a new index file under `docs/index.html`  just to redirect to the most recent dokka entry point
- Updated the README with the new link and a process to release a new version.
- Updated the [project GitHub config](https://github.com/Automattic/Gravatar-SDK-Android/settings/pages) to use the old school way of managing pages. I'll point it to `trunk` once this is merged.
- Removed the GitHub workflow to generate Dokka docs.

Note: I didn't push 0.1.0 or 0.2.0 docs version because: 1. it's not useful (and will never be useful) 2. it needs to be generated with the versioning plugin to work correctly so I'd have to back port this to 0.1.0 and 0.2.0.

### Testing Steps

Since there is only one version, if you visit the docs directory or visit [github pages](https://automattic.github.io/Gravatar-SDK-Android/dokka/current/index.html), you won't see a dropdown selector, you'll only see the version next to the title.

To test it:
- bump `versionName` to 0.4.0 in the root `build.gradle.kts`
- run `./gradlew dokkaHtmlMultiModule`
- verify there is a `docs/dokka/history/0.4.0/` directory 
- open `docs/dokka/current/index.html` and check you see the dropdown menu, try to navigate to 0.3.0 and check everything works as expected.

<img width="337" alt="Screenshot 2024-05-15 at 16 58 59" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/40213/2549c1c9-7992-493a-b1f8-7ebfaf4f3d4d">
